src/client/api/event.c

```diff
@@ -1222,12 +1222,10 @@ daos_event_priv_get(daos_event_t **ev)
 
 	D_ASSERT(*ev == NULL);
 
-	if (!ev_thpriv_is_init) {
-		rc = daos_event_priv_reset();
-		if (rc)
-			return rc;
-		ev_thpriv_is_init = true;
-	}
+	rc = daos_event_priv_reset();
+	if (rc)
+		return rc;
+	ev_thpriv_is_init = true;
```

Comment:
daos_event_priv_get() possibly be called at any time even when the event is in use and not finished using/polled out?
If you always do the reset for it, looks possibly break some use cases.

"It is possible that the error is not reset to 0 when a process is
retrieving the private event in a multi-threaded environment since
another thread might have processed the event completion."
just note that the priv event is per thread, so different threads are using different instances.

Reply:
for a single thread, aren't you always blocked in the wait for the private event before you call daos_event_priv_get() for the same event for that thread? im not sure how daos_event_priv_get can be called anytime. let me check this more.

"just note that the priv event is per thread, so different threads are using different instances."
yes but it is possible that another thread can mark the event as complete through the task completion in tse, but the thread that the event belongs to never resets the event in this case?

Reply:
in my testing im seeing a segfault in the event completion, so this is still not right. probably related to what you were saying.

Reply:
so it sounds like there is a race in daos_event_complete_locked and ev_progress_cb between different threads on the private event. even though the event itself is private, another thread can actually call daos_event_complete() on it.

so in daos_event_complete_locked the non-owner thread sets the status to completed, and sets the error code; then the owner thread of the private event marks the event as ready since it was marked completed, and resets it; while the other thread gets around to set the actual error code causing the resetted event to have a non zero error code.